centreon/www/front_src/src/Resources/Actions/Resource/Check/index.tsx

```diff
@@ -0,0 +1,186 @@
+import { SetStateAction, useEffect, useState } from 'react';
+
+import { isNil } from 'ramda';
+import { useTranslation } from 'react-i18next';
+
+import IconForcedCheck from '@mui/icons-material/FlipCameraAndroidOutlined';
+import IconArrow from '@mui/icons-material/KeyboardArrowDownOutlined';
+import IconCheck from '@mui/icons-material/Sync';
+
+import { Method, useMutationQuery, useSnackbar } from '@centreon/ui';
+
+import { Resource } from '../../../models';
+import {
+  labelCheck,
+  labelCheckCommandSent,
+  labelForcedCheck,
+  labelForcedCheckCommandSent
+} from '../../../translatedLabels';
+import { checkEndpoint } from '../../api/endpoint';
+import useAclQuery from '../aclQuery';
+import ResourceActionButton from '../ResourceActionButton';
+
+import { adjustCheckedResources } from './helpers';
+import ListCheckOptions from './ListCheckOptions';
+
+interface DisplaySelectedOption {
+  checked: boolean;
+  forcedChecked: boolean;
+}
+
+interface Props {
+  selectedResources: Array<Resource>;
+  setSelectedResources: (update: SetStateAction<Array<Resource>>) => void;
+  testId: string;
+}
+
+const CheckActionButton = ({
+  selectedResources,
+  setSelectedResources,
+  testId
+}: Props): JSX.Element => {
+  const { t } = useTranslation();
+
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+  const [displaySelectedOption, setDisplaySelectedOption] =
+    useState<DisplaySelectedOption | null>(null);
+  const { mutateAsync: checkResource } = useMutationQuery({
+    getEndpoint: () => checkEndpoint,
+    method: Method.POST
+  });
+
+  const { canForcedCheck, canCheck } = useAclQuery();
+  const { showSuccessMessage } = useSnackbar();
+
+  const disableCheck = !canCheck(selectedResources);
+  const disableForcedCheck = !canForcedCheck(selectedResources);
+  const hasSelectedResources = selectedResources.length > 0;
+
+  const isCheckPermitted = canCheck(selectedResources) || !hasSelectedResources;
+  const isForcedCheckPermitted =
+    canForcedCheck(selectedResources) || !hasSelectedResources;
+
+  const displayPopover = (event: React.MouseEvent<HTMLElement>): void => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const closePopover = (): void => {
+    setAnchorEl(null);
+  };
+
+  const handleCheckResource = ({ checked, forcedChecked }): void => {
+    setDisplaySelectedOption({
+      checked,
+      forcedChecked
+    });
+    setAnchorEl(null);
+    if (selectedResources.length <= 0) {
+      return;
+    }
```

Comment:
I think this condition is not relevant because the button should be disabled when there is no resource selected. wdyt?